signs-work-order-map/utils/iFrameMessenger.tsx

```diff
@@ -0,0 +1,33 @@
+"use client";
+import { useEffect, useState } from "react";
+
+type MessageToIframe = {
+  message: string;
+  records: [];
+  location: [number, number];
+};
+
+export function useIFrameMessenger() {
+  const [message, setMessage] = useState<MessageToIframe | null>(null);
+
+  useEffect(() => {
+    const consoleMessage = (event: MessageEvent) => {
+      if (
+        event.data.source === "react-devtools-content-script" ||
+        event.data.source === "react-devtools-bridge" ||
+        event.data.source === "react-devtools-backend-manager"
+      ) {
+        return;
+      }
+      const data = JSON.parse(event.data);
```

Comment:
i'm not sure how much we want to concern ourselves with this, but data is typed as `any` here and we are assuming that it conforms to the `MessageToIframe` type. we could add type guards here if we wanted to 🤷